providers/gitlab.go

```diff
@@ -183,6 +186,67 @@ func (p *GitLabProvider) verifyGroupMembership(userInfo *gitlabUserInfo) error {
 	return fmt.Errorf("user is not a member of '%s'", p.Group)
 }
 
+type gitlabProjectInfo struct {
+	Name     string `json:"name"`
+	Archived bool   `json:"archived"`
+}
+
+func (p *GitLabProvider) getProjectInfo(ctx context.Context, s *sessions.SessionState, project string) (*gitlabProjectInfo, error) {
+	endpointURL := &url.URL{
+		Scheme: p.LoginURL.Scheme,
+		Host:   p.LoginURL.Host,
+		Path:   "/api/v4/projects/",
+	}
+	endpoint := endpointURL.String() + url.QueryEscape(project)
+	req, err := http.NewRequestWithContext(ctx, "GET", endpoint, nil)
```

Comment:
Please use the request builder from `github.com/oauth2-proxy/oauth2-proxy/pkg/requests` as can be seen in other providers

https://github.com/oauth2-proxy/oauth2-proxy/blob/37026b60cebd89e2ae3aa489447e0038ca50895c/providers/github.go#L131-L138

I think this should work

```
err := requests.New(endpoint). 
 	WithContext(ctx). 
 	SetHeader("Authorization", "Bearer "+s.AccessToken). 
 	Do(). 
 	UnmarshalInto(&projectInfo) 
 if err != nil { 
 	return false, err 
 } 
```